### PR TITLE
Upgrade terraform-provider-azurerm to v3.67.0

### LINKED ISSUE #1319: Upgrade terraform-provider-azurerm to v3.67.0
Release details: https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.67.0

### PR DESCRIPTION
**Pull request by aq17**
This PR was generated via `$ upgrade-provider pulumi/pulumi-azure`.

---

- Upgrading terraform-provider-azurerm from 3.66.0  to 3.67.0.
	Fixes #1319
- Upgrading pulumi-terraform-bridge from v3.54.1 to v3.54.3.

### CODE REVIEW

File: provider/resource.go [hunk 1]
```diff
@@ -1760,7 +1762,6 @@ func Provider() tfbridge.ProviderInfo {
 			},
 
 			// Network
-			"azurerm_network_function_azure_traffic_collector": {Tok: azureResource(azureNetwork, "FunctionAzureTrafficController")},
```
Comment (aq17) [inline provider/resource.go]: Based on the upstream docs, looks like this new resource belongs in a new module `network_function`, not `network`

**Review by mjeffryes — APPROVED** [review]
👍

File: sdk/go/azure/network/functionAzureTrafficController.go [hunk 2]
```diff
@@ -60,6 +60,8 @@ import (
 //	$ pulumi import azure:network/functionAzureTrafficController:FunctionAzureTrafficController example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourceGroup1/providers/Microsoft.NetworkFunction/azureTrafficCollectors/azureTrafficCollector1
 //
 // ```
+//
+// Deprecated: azure.network/functionazuretrafficcontroller.FunctionAzureTrafficController has been deprecated in favor of azure.networkfunction/azuretrafficcollector.AzureTrafficCollector
```
Comment (aq17) [inline sdk/go/azure/network/functionAzureTrafficController.go:64]: Ah looks it interprets the change as a deprecation – I'll reset both schema gen commits and build from scratch

File: provider/cmd/pulumi-resource-azure/bridge-metadata.json [hunk 3]
```diff
@@ -15035,6 +15048,25 @@
...
+                "majorVersion": 5,
+                "fields": {
+                    "collector_policy_ids": {
+                        "maxItemsOne": false
+                    },
+                    "virtual_hub_id": {
+                        "maxItemsOne": false
+                    }
+                }
+            },
```
Comment (iwahbe) [inline provider/cmd/pulumi-resource-azure/bridge-metadata.json:15069]: To change the token from `azure:network/functionAzureTrafficController:FunctionAzureTrafficController` to `azure:networkFunction/azureTrafficController:AzureTrafficController` without introducing a deprecated resource, you will need to remove this block and regenerate (`make tfgen build_sdks`).

**Pull request merged**